companion/flashing.md

```diff
@@ -41,43 +63,77 @@ Zadig will install the driver, and should report success. When done you can dism
 
 ```
 sudo apt-get install dfu-util
 ```
 
-### Setup for Mac OS X
+### Flashing Using A Mac OS
 
 Companion for Mac OS already includes the required \`\`dfu-util'' tool  
 and does not require any additional installation of tools or drivers.
 
-## Downloading and flashing the firmware
-
-In OpenTX companion's preferences window, make sure the proper  
-firmware is selected \(OpenTX for FrSky Taranis \(X9D\), or OpenTX for  
-Taranis Plus \(X9D+\) according to the hardware you have\), and the  
-language / options match your preference, then click the top Download  
-button and select a location for the file. This will automatically  
-build and download the latest available version of OpenTX for Taranis.  
-Close the Preferences window, and select the Burn -&gt; Write Flash  
-memory command. Make sure your radio is powered off and connected.  
-Browse for the file you just downloaded. It will be recognised  
-automatically and the version will be displayed.  If you wish to  
-replace the radio's splash screen, you can either load one from a file  
-or select it from the available library, and click the "Substitute  
-image in firmware" box.  Click "Burn to TX".  Once the progress bar is  
-done and the success prompt is shown you can dismiss it and disconnect  
-your radio :\) If you get an error about the dfu-util executable not  
-being found, open the flashing tool settings with the Burn -&gt;  
-Configure menu item. Browse for the dfu-util executable, which should  
-be: On Windows: In the OpenTX companion installation folder \(by  
-default C:\Program Files\companion9x\dfu-util.exe on 32-bit systems,  
-and C:\Program Files \(x86\)\companion9x\dfu-util.exe on 64-bit  
-systems\).  On Mac OS: `/Applications/OpenTX Companion  
-2.2.app/Contents/Resources/dfu-util`
+## Downloading The OpenTX Firmware
+
+The only recommended way to download the OpenTX firmware is using Companion. This process compiles/builds the firmware on the OpenTX servers and downloads a copy to the computer.
+
+To download the firmware follow these steps:
+
+* Radio Profile: Make sure the correct transmitter profile is selected
+* Build Options: Make sure the build options selected are the ones you want
+* Download Firmware: File -> Download
+* Click on Download firmware (The download could take a few minutes)
+* Save the firmware .bin file on the computer
+* You may be prompted to write the firmware to the transmitter
+* If you want to choose yes connect the transmitter to the computer in bootloader mode
+* Click on Yes
+* Select Check Hardware compatibility to make sure the firmware being written to the transmitter is for that particular hardware.
+* Click on Write to TX
+* Eject the mounted drives
+* Disconnect the USB cable
+* Power the transmitter off
+
+# Writing The OpenTX Firmware
+
+Connect the transmitter to the computer in bootloader mode.
+
+There are two options to write the firmware:
+* Using The Bootloader Menu
+* Using Companion
```

Comment:
This is not accurate!

First using `dfu-util` on Windows requires Zadig driver. So they are tied together. On Linux/MAC there is no need for Zadig driver (since driver is built into the kernel), only dfu-util is needed. 

Flashing from Companion, these are possible ways:
 * radio off - connect to computer - you are in STM Bootloader - Companion auto detects this and uses zadig/dfu-util to flash
 * radio on with trim held inwards - you are in OpenTX Bootloader, connect to computer. Now you have two sub choices:
   * either flash directly from Companion -  this will work without any drivers, the bin file is copied to special file (using mass storage USB mode) that represents the flash of the radio. 
   * or copy the file on the SD card (via file manager), disconnect radio - use menu in OpenTX Bootloader to flash it.

Then there is also procedure to update the OpenTX Bootloader itself. It is similar to the last option above, just the flashing is done from the regular OpenTX, using the same file.

Reply:
So what is the goal of this section?
For me the goal is to guide users to get things done; not list every single technical way things can be done. These extra eays can be in an advanced trouble shooting section. 

When ist says there are two options to write the firmware it is under the sentence connect the transmitter to the computer in boot loader mode. Then there will be a boot loader screen image and then the rest of that text.

Reply:
It does not need to explain every detail. But what it explains should be correct.

Reply:
Agreed...
So when the Tx is turned on in boot loader moder this screen appears
![companion-bootloader](https://user-images.githubusercontent.com/20209851/30508150-f31cff28-9a5d-11e7-8d1c-aabd47747cb6.png)
Now there are two choices
Write the firmeware using that menu
Plug in the usb cable and use companion to write the firmware
What do I need to change?

Reply:
We could add a separate section on boot loader... what it is and the ways it get updated...